src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -666,52 +664,55 @@ int ObjectSynchronizer::wait(Handle obj, jlong millis, TRAPS) {
   return ret_code;
 }
 
-void ObjectSynchronizer::wait_uninterruptibly(Handle obj, jlong millis, TRAPS) {
+// No exception are possible in this case as we only use this internally when locking is
+// correct and we have to wait until notified - so no interrupts or timeouts.
+void ObjectSynchronizer::wait_uninterruptibly(Handle obj, JavaThread* current) {
   if (UseBiasedLocking) {
-    BiasedLocking::revoke(obj, THREAD);
+    BiasedLocking::revoke(obj, current);
     assert(!obj->mark().has_bias_pattern(), "biases should be revoked by now");
   }
-  if (millis < 0) {
-    THROW_MSG(vmSymbols::java_lang_IllegalArgumentException(), "timeout value is negative");
-  }
   // The ObjectMonitor* can't be async deflated because the _waiters
   // field is incremented before ownership is dropped and decremented
   // after ownership is regained.
-  ObjectMonitor* monitor = inflate(THREAD, obj(), inflate_cause_wait);
-  monitor->wait(millis, false, THREAD);
+  ObjectMonitor* monitor = inflate(current, obj(), inflate_cause_wait);
+  monitor->wait(0 /* wait-forever */, false /* not interruptible */, current);
```

Comment:
Didn't expect to see this change from "millis" to "0" either.
Seems out of context for this change.

Update: I see that you deleted the 'millis' param now. I missed that before.